commit 1ed6e300d57a48129786ba2dfbda6da1e233a454 (HEAD -> wip/T23415-9.8)
Author: Ian-Woo Kim <ianwookim@gmail.com>
Date:   Mon May 22 12:22:33 2023 -0700

    Add missing BCO handling in scavenge_one.
    
    (cherry picked from commit 902ebcc2b95707319d37a19d6b23c342cc14b162)

diff --git a/rts/sm/Scav.c b/rts/sm/Scav.c
index 8debec6a666..9bbe069d875 100644
--- a/rts/sm/Scav.c
+++ b/rts/sm/Scav.c
@@ -1593,6 +1593,14 @@ scavenge_one(StgPtr p)
 #endif
       break;
 
+    case BCO: {
+        StgBCO *bco = (StgBCO *)p;
+        evacuate((StgClosure **)&bco->instrs);
+        evacuate((StgClosure **)&bco->literals);
+        evacuate((StgClosure **)&bco->ptrs);
+        break;
+    }
+
     case COMPACT_NFDATA:
         scavenge_compact((StgCompactNFData*)p);
         break;
